jdbc-adapter/virtualschema-jdbc-adapter/src/test/java/com/exasol/adapter/dialects/impl/PostgreSQLSqlDialectTest.java

```diff
@@ -0,0 +1,67 @@
+package com.exasol.adapter.dialects.impl;
+
+import com.exasol.adapter.dialects.SqlDialectContext;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import static org.mockito.Mockito.*;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+public class PostgreSQLSqlDialectTest {
+    @Mock
+    SqlDialectContext sqlDialectContext;
+
+
+    @Before
+    public void setUp() throws SQLException {
+
+    }
+
+    @Test
+    public void applyQuoteOnUpperCase() {
+        PostgreSQLSqlDialect postgresDialect = new PostgreSQLSqlDialect(sqlDialectContext);
+        assertEquals("\"abc\"", postgresDialect.applyQuote("ABC"));
+    }
+
+    @Test
+    public void applyQuoteOnMixedCase() {
+        PostgreSQLSqlDialect postgresDialect = new PostgreSQLSqlDialect(sqlDialectContext);
+        assertEquals("\"abcde\"", postgresDialect.applyQuote("AbCde"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void mapTableWithUpperCaseCharacters() throws SQLException {
```

Comment:
Improve test case name: `testMapTableWithUpperCaseCharactersAndNoErrorIgnoredThrowsException`
Don't be worried about length of the method name. Sometimes precision needs space.

Reply:
Extract common part of this and the following two methods to reduce code duplication.

Reply:
extract not possible because of mocking